modules/hivemq-edge-module-etherip/src/test/java/com/hivemq/edge/adapters/etherip/config/EipAdapterConfigTest.java

```diff
@@ -0,0 +1,204 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hivemq.edge.adapters.etherip.config;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.hivemq.adapter.sdk.api.config.MqttUserProperty;
+import com.hivemq.configuration.entity.HiveMQConfigEntity;
+import com.hivemq.configuration.reader.ConfigFileReaderWriter;
+import com.hivemq.configuration.reader.ConfigurationFile;
+import com.hivemq.edge.adapters.etherip.EipProtocolAdapterFactory;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+
+import static com.hivemq.adapter.sdk.api.config.MessageHandlingOptions.MQTTMessagePerSubscription;
+import static com.hivemq.adapter.sdk.api.config.MessageHandlingOptions.MQTTMessagePerTag;
+import static com.hivemq.protocols.ProtocolAdapterUtils.createProtocolAdapterMapper;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+class EipAdapterConfigTest {
+
+    private final @NotNull ObjectMapper mapper = createProtocolAdapterMapper(new ObjectMapper());
+
+    @Test
+    public void convertConfigObject_fullConfig_valid() throws Exception {
+        final URL resource = getClass().getResource("/eip-adapter-full-config.xml");
+        final File path = Path.of(resource.toURI()).toFile();
+
+        final HiveMQConfigEntity configEntity = loadConfig(path);
+        final Map<String, Object> adapters = configEntity.getProtocolAdapterConfig();
+
+        final EipProtocolAdapterFactory eipProtocolAdapterFactory = new EipProtocolAdapterFactory();
+        final EipAdapterConfig config =
+                eipProtocolAdapterFactory.convertConfigObject(mapper, (Map) adapters.get("eip"));
+
+        assertThat(config.getId()).isEqualTo("my-eip-protocol-adapter");
+        assertThat(config.getPort()).isEqualTo(1234);
+        assertThat(config.getHost()).isEqualTo("my.eip-server.com");
+        assertThat(config.getBackplane()).isEqualTo(4);
+        assertThat(config.getSlot()).isEqualTo(5);
+        assertThat(config.getEipToMqttConfig().getPollingIntervalMillis()).isEqualTo(10);
+        assertThat(config.getEipToMqttConfig().getMaxPollingErrorsBeforeRemoval()).isEqualTo(9);
+        assertThat(config.getEipToMqttConfig().getPublishChangedDataOnly()).isFalse();
+        assertThat(config.getEipToMqttConfig().getMappings()).satisfiesExactly(mapping -> {
+            assertThat(mapping.getMqttTopic()).isEqualTo("my/topic");
+            assertThat(mapping.getMqttQos()).isEqualTo(1);
+            assertThat(mapping.getMessageHandlingOptions()).isEqualTo(MQTTMessagePerTag);
```

Comment:
```suggestion
            assertThat(mapping.getMessageHandlingOptions()).isEqualTo(MQTTMessagePerSubscription);
```
Do not test the default value here